utils/rewriter/rewriter.js

```diff
@@ -37,10 +25,17 @@ var ruleSets = null;
 function processDir(dir) {
   var stream = readdirp({
     root: dir,
-    fileFilter: ['*.html', '*.js', '*.rb', '*.erb', '*.mustache',
-      '*.scala', '*.c', '*.cc', '*.cpp', '*.cxx',
-      '*.java', '*.go', '*.php', '*.css', '*.pl', '*.py',
-      '*.rhtml', '*.sh', '*.yaml']
+    directoryFilter: [
+      '!.*', // Protect `.git` and other VCS directories.
+      '!*modules',
+      '!LICENSES', // https://reuse.software/ convention.
+    ],
+    fileFilter: [
+      '!*.png', '!*.svg', '!*.ico', '!*.xcf',
+      '!*.pdf',
+
+      '!LICENSE*', '!COPYING*', // Prefer https://reuse.software/.
+    ],
```

Comment:
I'd still prefer an allowlist to avoid any binary file. E.g. JPEG is also missing from the list.

Reply:
I am not really happy with that either. My idea was that I did not want to miss files. As I wrote, after the rewrite the diff will have to be reviewed anyway.

What do you think of using this https://github.com/gjtorikian/isBinaryFile instead?

Reply:
Drop it. https://github.com/gjtorikian/isBinaryFile is fruitless. Stick to a white list and use this tool carefully if you intend to use it as a sanitizer... I didn't review your code, but I don't understand why'd you remove the only documentation on top...

Reply:
> I didn't review your code, but I don't understand why'd you remove the only documentation on top...

I moved it to README.md so that it will be shown. Preview: https://github.com/ypid/https-everywhere/tree/fix/rewriter/utils/rewriter